docs/interactions.rst

```diff
@@ -0,0 +1,274 @@
+if you return a :class:`~pincer.objects.message.message.Message` object.
+
+.. code-block:: python
+
+  from pincer import Client, command, Embed
+  from pincer.objects import Message, File
+  ...
+
+  @command
+  async def a_complex_message(self):
+    return Message(
+      "This is the message's content"
+      embeds=[
+        Embed(
+          title="Pincer",
+          description=(
+            "🚀 An asynchronous python API wrapper meant to replace"
+            " discord.py\n> Snappy discord api wrapper written "
+            "with aiohttp & websockets"
+          ).set_image(
+            url="attachments://some_image.png"
+          )
+        )
+      ],
+      attachments=[
+        File.from_file("path/to/a/file.png", filename="new_name.png"),
+        "path/to/another/file.png" # A string is inferred to be a filepath here!
+      ]
+    )
+
+Attachments can also be Pillow images if Pillow is installed.
+
+.. code-block:: python
+
+  from PIL import Image
+  ...
+
+  attachments=[
+    Image.new("RGBA", (500, 500), (255, 0, 0)), # Will automatically be named `image0.png`
+    Image.new("RGB", (500, 500)), # Will automatically be named `image1.png`
+    File.from_pillow_image(some_pillow_image, "this_is_the_image_name.png") # You can also do this to specify the name
+  ]
+
+Additionally, Pillow Images, Files, and Embeds can be returned directly without wrapping them in a :class:`~pincer.objects.message.message.Message` object.
+
+.. code-block:: python
+
+  ...
+  @command
+  async def a_complex_message(self):
+    return Embed(
+      title="Pincer",
+      description=(
+        "🚀 An asynchronous python API wrapper meant to replace"
+        " discord.py\n> Snappy discord api wrapper written "
+        "with aiohttp & websockets"
+      )
+    )
+
+.. list-table:: Possible Return Types
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Return Type
+     - Discord Message
+   * - :class:`str`
+     - text only message
+   * - :class:`~pincer.objects.message.embed.Embed`
+     - Discord embed
+   * - :class:`~pincer.objects.message.file.File`
+     - file attachment
+   * - :class:`PIL.Image.Image`
+     - single image attachment
+
+
+Sending Messages Without Return
+-------------------------------
+The :class:`~pincer.objects.message.context.MessageContext` object provides methods to send a response to an interaction.
+
+.. code-block:: python
+
+    from pincer.objects import MessageContext, Message
+
+    @command
+    async def some_command(self, ctx: MessageContext):
+        ctx.send("Hello word!") # Sends hello world as the response to the interaction
+        return # No response will be sent now that the interaction has been completed
+
+    @command
+    async def some_other_command(self, ctx: MessageContext):
+        channel = await self.get_channel(ctx.channel_id)
+        channel.send("Hello word!") # Sends a message in the channel
+        return "Hello world 2" # This is sent because the interaction was not "used up"
```

Comment:
```suggestion
    async def some_command(self, ctx: MessageContext):
        ctx.send("Hello world!") # Sends hello world as the response to the interaction
        return # No response will be sent now that the interaction has been completed

    @command
    async def some_other_command(self, ctx: MessageContext):
        channel = await self.get_channel(ctx.channel_id)
        channel.send("Hello world!") # Sends a message in the channel
        return "Hello world 2" # This is sent because the interaction was not "used up"
```